

# `r fontawesome::fa("book", fill=" #4141f7")` : Anotación funcional y filogenia

![](img/Banner.jpg)

## PROKKA

```{bash, eval=FALSE}
$ conda create --yes -n prokka prokka
$ conda activate prokka

$ prokka --kingdom Bacteria --genus GENUS --species SPECIE --outdir annotation assembly/scaffolds.fasta

```

## Filogenia

```{bash, eval=FALSE}
$ conda create -n phylo blast mafft raxml iqtree bedtools

# Figure out the chromosome and location of the `gnd` gene
# there *must* be a better way to do this
grep -B 1 ‘NODE_\|gnd’ ../annotation/*gbk

# open a file to put the coordinates into using vi or nano
vi gnd.bed

# these are the coordinates from the contigs-file (Yours might be different), we copy into the vi/nano buffer
NODE_42_length_35862_cov_7.082632   625     2031

# safe the file and exit vi/nano

bedtools getfasta -fi ../assembly/contigs.fasta -bed gnd.bed > gnd.fasta

#finding orthologs
blastn -db nt -query gnd.fasta -remote -evalue 1e-100 -outfmt "6 qseqid sseqid sseq" > gnd_blast_hits.out

awk 'BEGIN { OFS = "\n" } { print ">"$2, $3 }' gnd_blast_hits.out > gnd_blast_hits.fasta

cat gnd.fasta >> gnd_blast_hits.fasta

#alignment
$ mafft gnd_blast_hits.fasta > gnd_blast_hits.aln

#building phylogeny

$ raxmlHPC -s gnd_blast_hits.aln -m GTRGAMMA -n ecoli_tree -p 12345
iqtree -s gnd_blast_hits.aln



```

